numalogic/connectors/druid.py

```diff
@@ -107,33 +112,36 @@ def fetch(
         _start_time = time.perf_counter()
         filter_pairs = make_filter_pairs(filter_keys, filter_values)
         query_params = build_params(
-            aggregations, datasource, dimensions, filter_pairs, granularity, hours
+            aggregations, datasource, dimensions, filter_pairs, granularity, hours, delay
         )
-
-        response = self.client.groupby(**query_params)
-        df = response.export_pandas()
-
-        if df is None or df.shape[0] == 0:
-            logging.warning("No data found for keys %s", filter_pairs)
+        try:
+            response = self.client.groupby(**query_params)
+        except Exception:
+            _LOGGER.exception("Problem with getting response from client")
             return pd.DataFrame()
-
-        df["timestamp"] = pd.to_datetime(df["timestamp"]).astype("int64") // 10**6
-
-        if group_by:
-            df = df.groupby(by=group_by).sum().reset_index()
-
-        if pivot.columns:
-            df = df.pivot(
-                index=pivot.index,
-                columns=pivot.columns,
-                values=pivot.value,
-            )
-            df.columns = df.columns.map("{0[1]}".format)
-            df.reset_index(inplace=True)
-
-        _end_time = time.perf_counter() - _start_time
-        _LOGGER.debug("Druid query latency: %.6fs", _end_time)
-        return df
+        else:
+            df = response.export_pandas()
+            if df is None or df.shape[0] == 0:
```

Comment:
Can df be None here? Or can we just check for df.empty?

Reply:
We needed the check for None before when there was no try and except block.